### DISCUSSION TITLE
An unexpected error occurred: Object reference not set to an instance of an object.

### DISCUSSION THREAD

**Discussion by robv12** [pszybiak/openapi-2-excel#2] [General]
what does error

An unexpected error occurred: Object reference not set to an instance of an object.

mean?

**Comment by pszybiak** [comment]
Thanks for reporting. Could you provide more details, such as a service definition?

This error is an internal error and I can't tell what's causing it based on the message alone.

**Comment by robv12** [comment]
Only details it displayed when i tried to convert a file . Is there a debug flag i can set to get more information?

Rob

On 20 Oct 2024, at 14:39, Paweł Szybiak ***@***.***> wrote:

﻿

Thanks for reporting. Could you provide more details, such as a service definition?

This error is an internal error and I can't tell what's causing it based on the message alone.

**Comment by pszybiak** [comment]
I have just published a new version of the tool with an added debug mode option (-g or --debug).
Please use it and paste the full text of the error that appears.

**Comment by robv12** [comment]
Just tried and only got the following

Error: Could not find color or style ‘T,TMinMax’



On 20 Oct 2024, at 15:50, Paweł Szybiak ***@***.***> wrote:

﻿

I have just published a new version of the tool with an added debug mode option (-g or --debug).
Please use it and paste the full text of the error that appears.

**Reply by pszybiak** [reply]
Is this a full message that is being presented?

**Reply by pszybiak** [reply]
I need the full text of the message, e.g. something like this (full text in red)

![image](https://github.com/user-attachments/assets/dd824db6-2e6e-4a77-97a2-5edd3cdcd888)

**Comment by robv12** [comment]
[cid:ff3a6f96-2882-4916-8b35-a0b42d34986d]

there is not anything else. checked for a file etc but above is the only text i get
________________________________

**Reply by pszybiak** [reply]
Thanks for the clarification. 
I've released another version with improved error logging. Should you update the tool and try again?

**Comment by robv12** [comment]
An unexpected error occurred: System.NullReferenceException: Object reference not set to an instance of an object.
   at openapi2excel.core.Builders.WorksheetPartsBuilders.Common.MaxPropertiesTreeLevel.EstablishMaxTreeLevel(OpenApiSchema schema, Int32 currentLevel, Int32
maxTreeLevel) in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/Common/MaxPropertiesTreeLevel.cs:line
38
   at openapi2excel.core.Builders.WorksheetPartsBuilders.Common.MaxPropertiesTreeLevel.<>c__DisplayClass1_0.<EstablishMaxTreeLevel>b__1(OpenApiSchema
schema) in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/Common/MaxPropertiesTreeLevel.cs:line 25
   at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.AppendPrependIterator`1.LoadFromEnumerator()
   at System.Linq.Enumerable.AppendPrepend1Iterator`1.MoveNext()
   at System.Linq.Enumerable.MinMaxInteger[T,TMinMax](IEnumerable`1 source)
   at openapi2excel.core.Builders.WorksheetPartsBuilders.Common.MaxPropertiesTreeLevel.EstablishMaxTreeLevel(IDictionary`2 mediaTypes, Int32 maxTreeLevel)
in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/Common/MaxPropertiesTreeLevel.cs:line 24
   at openapi2excel.core.Builders.WorksheetPartsBuilders.Common.MaxPropertiesTreeLevel.<>c__DisplayClass0_0.<Calculate>b__0(KeyValuePair`2 response) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/Common/MaxPropertiesTreeLevel.cs:line 18
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.AppendPrependIterator`1.LoadFromEnumerator()
   at System.Linq.Enumerable.AppendPrepend1Iterator`1.MoveNext()
   at System.Linq.Enumerable.MinMaxInteger[T,TMinMax](IEnumerable`1 source)
   at openapi2excel.core.Builders.WorksheetPartsBuilders.Common.MaxPropertiesTreeLevel.Calculate(OpenApiOperation operation, Int32 maxTreeLevel) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/Common/MaxPropertiesTreeLevel.cs:line 18
   at openapi2excel.core.Builders.OperationWorksheetBuilder.Build(String path, OpenApiPathItem pathItem, OperationType operationType, OpenApiOperation
operation) in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/OperationWorksheetBuilder.cs:line 24
   at openapi2excel.core.OpenApiDocumentationGenerator.<>c__DisplayClass2_1.<GenerateDocumentationImpl>b__1(KeyValuePair`2 operation) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 48
   at openapi2excel.core.Common.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Common/EnumerableExtensions.cs:line 9
   at openapi2excel.core.OpenApiDocumentationGenerator.<>c__DisplayClass2_0.<GenerateDocumentationImpl>b__0(KeyValuePair`2 path) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 45
   at openapi2excel.core.Common.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Common/EnumerableExtensions.cs:line 9
   at openapi2excel.core.OpenApiDocumentationGenerator.GenerateDocumentationImpl(Stream openApiFileStream, String outputFile, OpenApiDocumentationOptions
options) in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 44
   at openapi2excel.core.OpenApiDocumentationGenerator.GenerateDocumentation(String openApiFile, String outputFile, OpenApiDocumentationOptions options) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 21
   at openapi2excel.core.OpenApiDocumentationGenerator.GenerateDocumentation(String openApiFile, String outputFile, OpenApiDocumentationOptions options) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 21
   at OpenApi2Excel.cli.GenerateExcelCommand.Execute(CommandContext context, GenerateExcelSettings settings) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel.cli/GenerateExcelCommand.cs:line 102
________________________________

**Reply by pszybiak** [reply]
Thank you very much.
I have uploaded a fix. I hope it helps. If not, please provide more logs.

**Comment by robv12** [comment]
An unexpected error occurred: System.NullReferenceException: Object reference not set to an instance of an object.
   at openapi2excel.core.Builders.WorksheetPartsBuilders.PropertiesTreeBuilder.CorrectRootElementIfArray(OpenApiSchema schema) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/PropertiesTreeBuilder.cs:line 49
   at openapi2excel.core.Builders.WorksheetPartsBuilders.PropertiesTreeBuilder.AddPropertiesTree(RowPointer actualRow, OpenApiSchema schema,
OpenApiDocumentationOptions options) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/PropertiesTreeBuilder.cs:line 41
   at openapi2excel.core.Builders.WorksheetPartsBuilders.PropertiesTreeBuilder.AddPropertiesTreeForMediaTypes(RowPointer actualRow, IDictionary`2
mediaTypes, OpenApiDocumentationOptions options) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/PropertiesTreeBuilder.cs:line 30
   at openapi2excel.core.Builders.WorksheetPartsBuilders.ResponseBodyBuilder.AddResponseBodyPart(OpenApiOperation operation) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/WorksheetPartsBuilders/ResponseBodyBuilder.cs:line 28
   at openapi2excel.core.Builders.OperationWorksheetBuilder.AddResponseBody(OpenApiOperation operation) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/OperationWorksheetBuilder.cs:line 108
   at openapi2excel.core.Builders.OperationWorksheetBuilder.Build(String path, OpenApiPathItem pathItem, OperationType operationType, OpenApiOperation
operation) in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Builders/OperationWorksheetBuilder.cs:line 31
   at openapi2excel.core.OpenApiDocumentationGenerator.<>c__DisplayClass2_1.<GenerateDocumentationImpl>b__1(KeyValuePair`2 operation) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 48
   at openapi2excel.core.Common.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Common/EnumerableExtensions.cs:line 9
   at openapi2excel.core.OpenApiDocumentationGenerator.<>c__DisplayClass2_0.<GenerateDocumentationImpl>b__0(KeyValuePair`2 path) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 45
   at openapi2excel.core.Common.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/Common/EnumerableExtensions.cs:line 9
   at openapi2excel.core.OpenApiDocumentationGenerator.GenerateDocumentationImpl(Stream openApiFileStream, String outputFile, OpenApiDocumentationOptions
options) in /home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 44
   at openapi2excel.core.OpenApiDocumentationGenerator.GenerateDocumentation(String openApiFile, String outputFile, OpenApiDocumentationOptions options) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 21
   at openapi2excel.core.OpenApiDocumentationGenerator.GenerateDocumentation(String openApiFile, String outputFile, OpenApiDocumentationOptions options) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel/OpenApiDocumentationGenerator.cs:line 21
   at OpenApi2Excel.cli.GenerateExcelCommand.Execute(CommandContext context, GenerateExcelSettings settings) in
/home/runner/work/openapi-2-excel/openapi-2-excel/src/openapi2excel.cli/GenerateExcelCommand.cs:line 102
________________________________

**Reply by pszybiak** [reply]
Sorry it looks like this, but without the source file I can't fix all the errors right away. 
I've uploaded another version for verification.

**Comment by robv12** [comment]
It worked 🙂.

Will check the excel produced
________________________________

**Comment by robv12** [comment]
Thank you so much for the update

On 21 Oct 2024, at 12:50, Paweł Szybiak ***@***.***> wrote:

﻿

Sorry it looks like this, but without the source file I can't fix all the errors right away.
I've uploaded another version for verification.

**Comment by robv12** [comment]
Hi,

  Come across two other different errors.

  added the yaml files and the output.


________________________________

**Comment by pszybiak** [comment]
Thanks for the additional feedback. I see the error messages, but unfortunately, I don't see the YAML files, and they would be very helpful in diagnosing the issues.

**Comment by robv12** [comment]
Do you see attached files or are they being removed
________________________________

**Comment by robv12** [comment]
see if they get through when zipped
________________________________

**Comment by pszybiak** [comment]
I don't see any attachment.
What is the size of the attached files? According to the documentation, they cannot exceed 10 MB.
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files

**Reply by pszybiak** [reply]
Test: attaching zip file
[openapi.zip](https://github.com/user-attachments/files/17520320/openapi.zip)

**Reply by pszybiak** [reply]
Do you see a link to file?

**Comment by robv12** [comment]
Yes got your file. But seems like mine dont get through. Let me try again[https://res.cdn.office.net/assets/mail/file-icon/png/zip_16x16.png]openapi_2.zip<https://1drv.ms/u/c/f12d69648a7cbaea/Ec89tToEdwNBmrGOZ1iE5BMB0sNRuSjjt5UU2ttr_Q0iww>
________________________________

**Comment by robv12** [comment]
Added .txt to see if that works
________________________________

**Comment by pszybiak** [comment]
I see text file and link to OneDrive also works. Weird.
I will analyze the documents provided and try to respond as soon as possible.

**Comment by pszybiak** [comment]
Ok, I've reviewed the provided attachments. The errors that appeared during the loading of the first one have been fixed. **You can download the version and check it.**

Unfortunately, there are more issues with the second file, as it has several errors and therefore is not a valid file according to the [ OpenAPI Specification](https://swagger.io/specification).

For example:

- Incorrect example addresses: instead of `#components/examples`, it should be `#/components/examples`.

- The following examples are missing in the `#components/examples` section:
  - `#/components/examples/externalCallRdo`
  - `#/components/examples/listaliases`
- Non-unique addresses, e.g., the `delete` operation is performed at the following addresses:
  - `/services/private/{id}`
  - `/services/private/{typeId}`
